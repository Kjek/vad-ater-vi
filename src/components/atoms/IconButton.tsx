import CancelIcon from '@asset/CancelIcon';
import { cn } from '@util/cn';
import type { ButtonHTMLAttributes } from 'react';

interface IconButtonProps extends ButtonHTMLAttributes<HTMLButtonElement> {
  variant: 'cancel';
}

const IconButton = (props: IconButtonProps) => {
  const { variant, className } = props;
  const icon = () => {
    switch (variant) {
      case 'cancel':
        return <CancelIcon />;
      default:
        break;
    }
  };

  return (
    <button
      {...props}
      type='button'
      className={cn(
        className,
        'rounded-full hover:bg-gray-700 dark:hover:bg-gray-700'
      )}
    >
      {icon()}
    </button>
  );
};

export default IconButton;
