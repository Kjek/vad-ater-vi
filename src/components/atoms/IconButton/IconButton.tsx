import { CopyIcon, Cross1Icon } from '@radix-ui/react-icons';
import { cn } from '@util/cn';
import type { ComponentPropsWithoutRef } from 'react';

interface IconButtonProps extends ComponentPropsWithoutRef<'button'> {
  variant: 'cancel' | 'copy';
}

const IconButton = ({ variant, className, ...props }: IconButtonProps) => {
  const icon = () => {
    switch (variant) {
      case 'cancel':
        return <Cross1Icon className='text-red-500 dark:text-red-700' />;
      case 'copy':
        return <CopyIcon className='text-gray-500 dark:text-gray-300' />;
      default:
        break;
    }
  };

  return (
    <button
      {...props}
      type='button'
      className={cn(
        'rounded-full p-2 hover:bg-gray-200 dark:hover:bg-gray-700',
        className
      )}
    >
      {icon()}
    </button>
  );
};

export default IconButton;
