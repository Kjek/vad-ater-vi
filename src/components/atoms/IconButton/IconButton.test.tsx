import { CopyIcon, Cross1Icon } from '@radix-ui/react-icons';
import { render, screen } from '@testing-library/react';
import React from 'react';
import { describe, expect, it } from 'vitest';
import IconButton from './IconButton';

describe(IconButton, () => {
  it('renders correctly', () => {
    const { getByTestId } = render(
      <IconButton data-testid={'icon.button'} variant='cancel' />
    );
    const button = getByTestId('icon.button') as HTMLButtonElement;
    expect(button).toBeInTheDocument();
    expect(button).toBeVisible();
  });

  it('displays cancel correctly', () => {
    const { getByTestId } = render(
      <IconButton data-testid={'icon.button'} variant='cancel' />
    );
    const button = getByTestId('icon.button') as HTMLButtonElement;
    expect(button).toBeVisible();
    screen.debug();

    const { container: Cross1IconContainer } = render(<Cross1Icon />);

    expect(button).toHaveTextContent(Cross1IconContainer.textContent ?? '');
  });

  it('displays copy correctly', () => {
    const { getByTestId } = render(
      <IconButton data-testid={'icon.button'} variant='copy' />
    );
    const button = getByTestId('icon.button') as HTMLButtonElement;
    expect(button).toBeVisible();
    screen.debug();

    const { container: CopyIconContainer } = render(<CopyIcon />);

    expect(button).toHaveTextContent(CopyIconContainer.textContent ?? '');
  });
});
