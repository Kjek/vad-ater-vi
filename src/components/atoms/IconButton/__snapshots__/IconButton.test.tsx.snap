// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`IconButton > matches snapshot 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <button
        class="rounded-full p-2 hover:bg-gray-200 dark:hover:bg-gray-700"
        data-testid="icon.button"
        type="button"
      >
        <div>
          CopyIcon
        </div>
      </button>
    </div>
  </body>,
  "container": <div>
    <button
      class="rounded-full p-2 hover:bg-gray-200 dark:hover:bg-gray-700"
      data-testid="icon.button"
      type="button"
    >
      <div>
        CopyIcon
      </div>
    </button>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`IconButton > matches snapshot 2`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <button
        class="rounded-full p-2 hover:bg-gray-200 dark:hover:bg-gray-700"
        data-testid="icon.button"
        type="button"
      >
        <div>
          CopyIcon
        </div>
      </button>
    </div>
    <div>
      <button
        class="rounded-full p-2 hover:bg-gray-200 dark:hover:bg-gray-700"
        data-testid="icon.button"
        type="button"
      >
        <div>
          Cross1Icon
        </div>
      </button>
    </div>
  </body>,
  "container": <div>
    <button
      class="rounded-full p-2 hover:bg-gray-200 dark:hover:bg-gray-700"
      data-testid="icon.button"
      type="button"
    >
      <div>
        Cross1Icon
      </div>
    </button>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
